src/components/prendus-questions/prendus-questions-view.ts

```diff
@@ -0,0 +1,141 @@
+import {SetPropertyAction, SetComponentPropertyAction, DefaultAction} from '../../prendus.d';
+import {User, Question, Assignment} from '../../prendus.d';
+import {createUUID, navigate, fireLocalAction, asyncForEach} from '../../node_modules/prendus-shared/services/utilities-service';
+import {shuffleArray} from '../../services/utilities-service';
+import {QuestionType, NotificationType, ContextType, VerbType, ObjectType} from '../../services/constants-service';
+import {setNotification, getAndSetUser, checkForUserToken} from '../../redux/actions';
+import {LTIPassback} from '../../services/lti-service';
+import {sendStatement} from '../../services/analytics-service';
+import {GQLRequest} from '../../node_modules/prendus-shared/services/graphql-service';
+
+export class PrendusQuestionsView extends Polymer.Element {
+  loaded: boolean;
+  action: SetPropertyAction | SetComponentPropertyAction | DefaultAction;
+  componentId: string;
+  userToken: string;
+  user: User;
+  userId: string;
+  noUserQuestions: boolean;
+  questionIds: string[];
+  questions:  Question[];
+  fetchQuestions: (pageIndex: number, pageAmount: number) => any[];
+  deleteItem: (questionId: string, userToken: string, questions: Question[]) => any[];
+
+  static get is() { return 'prendus-questions-view' }
+
+  static get properties() {
+    return{
+        userId: {
+        type: String,
+        observer: 'loadUser'
+      },
+      fetchQuestions: {
+        type: Function,
+        computed: 'computeFetchQuestions(user, userToken)'
+      },
+    }
+  }
+
+  constructor() {
+    super();
+    this.componentId = createUUID();
+  }
+
+  async connectedCallback() {
+    super.connectedCallback();
+    this.action = fireLocalAction(this.componentId, 'loaded', true);
+  }
+
+  async loadUser(){
+    this.action = checkForUserToken();
+    if(!this.user)
+      this.action = await getAndSetUser();
+  }
+  openConfirmDeleteQuestionModal(e){
+    this.shadowRoot.querySelector('#questionsList').shadowRoot.querySelector(`#${e.detail.questionId}`).open="false";
+    this.shadowRoot.querySelector('#questionsList').shadowRoot.querySelector(`#${e.detail.questionId}`).open="true";
+  }
+
+  computeFetchQuestions(user: User, userToken: string): (i: number, n: number) => Promise<object> | void {
+    //The load user function will check if the user is logged in. This ensures that the user is set before returning this function.
```

Comment:
So even if the user function doesn't get set first, we don't want to get the questions (and we can't) until the user is set. Maybe I should just delete the comment since it doesn't make sense to anyone else?